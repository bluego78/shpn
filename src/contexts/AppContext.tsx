/* IMPORT NODE MODULES & COMPONENTS*/
import {createContext} from 'react';

/* IMPORT INTERFACES */
import IUser from '../Interfaces/IUser';
import IAppContext from '../Interfaces/IAppContext';

let selectedNationCookie : string | null = localStorage.getItem("selectednat");

// Create the context
const appContext: IAppContext | any = createContext({ 
    isLoading: false,
    filterIsActive:false,
    usersList: [] as Array<IUser>,
    filteredUsersList: [] as Array<IUser>,
    currentPage: 0,
    totalResults:0,
    selectedNations: selectedNationCookie!==null ? selectedNationCookie.split(",") : [],
    modalIsOpened:false,
});

export default appContext;

