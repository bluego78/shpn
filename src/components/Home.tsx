/* IMPORT CSS */
import '../scss/App.scss';

/* IMPORT NODE MODULES & COMPONENTS */
import { useContext } from 'react';

/* IMPORT CONTEXTS */
import AppContext from '../contexts/AppContext';


/* IMPORT CUSTOM COMPONENTS */
import Loader from './Loader';
import UserList from './UserList';

export default (props:any) => {

    //Load the context
    const [appContext, setAppContext] = useContext(AppContext);
    
    return <div className="content-fluid d-flex flex-column align-items-center justify-content-center">
                {appContext.isLoading && appContext.currentPage===0 ? 
                <Loader />
                : 
                <UserList />
                }
            </div>
         
}
