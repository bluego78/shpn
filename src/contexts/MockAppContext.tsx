/* IMPORT NODE MODULES & COMPONENTS*/
import {useState} from 'react';
import AppContext from './AppContext';

export const fakeUsers = [
    {
        id: {value:"XX123"},
        login: { username:"theusername" },
        email:"test@example.com",
        name: {first:"John", last:"Doe"},
        nat:"ch",
        picture: { thumbnail:"https://randomuser.me/api/portraits/thumb/men/56.jpg" },
        phone:"+41 76 12345678",
        cell:"+41 76 36958488",
        location: {
                    city:"Chiasso", 
                    country:"Switzerland", 
                    postcode:"6830", 
                    state:"-",
                    street: {name:"C.so San Gottardo", number:"15"}
                }
    },
    {
        id: {value:"YY456"},
        login: { username:"anotherusername" },
        email:"test2@example.com",
        name: {first:"John", last:"Smith"},
        nat:"gb",
        picture: { thumbnail:"https://randomuser.me/api/portraits/thumb/men/57.jpg" },
        phone:"+44 12 44444444",
        cell:"+44 23 55555555",
        location: {
                    city:"London", 
                    country:"Britain", 
                    postcode:"1234", 
                    state:"-",
                    street: {name:"Par Street", number:"10"}
                }
    }
];

const fakeContext = {...AppContext, usersList: fakeUsers, filteredUsersList: fakeUsers, selectedNations:['ch'] };

export const MockedContextState = ({children}:any)=> {
    const [appState, setAppState] = useState(fakeContext);
    return <AppContext.Provider value={[appState, setAppState]}>{children}</AppContext.Provider>;
}

export const MockedContextStateLoading = ({children}:any)=> {
    let fakeContext = {...AppContext, isLoading:true, currentPage:0 };
    const [appState, setAppState] = useState(fakeContext);
    return <AppContext.Provider value={[appState, setAppState]}>{children}</AppContext.Provider>;
}


export const MockedContext = ({children}:any)=> {
    return <AppContext.Provider value={fakeContext}>{children}</AppContext.Provider>;
}

export const MockedContextLoading = ({children}:any)=> {
    let fakeContext = {...AppContext, isLoading:true, currentPage:0 };
    return <AppContext.Provider value={fakeContext}>{children}</AppContext.Provider>;
}


