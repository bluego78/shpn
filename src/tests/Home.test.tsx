import { MockedContextState, MockedContextStateLoading } from '../contexts/MockAppContext';
import Home from '../components/Home';
import renderer from 'react-test-renderer';

describe('<Home /> during the loading phase',()=>{

    let component:any;

    beforeAll(()=>{
        component = renderer.create(<MockedContextStateLoading><Home /></MockedContextStateLoading>);
    });

    afterAll(()=>{
        component = null;
    });

    it('Renders the app-container', ()=>{
        expect(component.root.findAllByProps({id: "home"}).length).toBe(1); 
    });
    
    it('Renders the <Loader /> component', ()=>{
        expect(component.root.findAllByProps({className: "loader"}).length).toBe(1); 
    });
    
});

describe('<Home /> when context is loaded',()=>{

    let component:any;

    beforeAll(()=>{
        component = renderer.create(<MockedContextState><Home /></MockedContextState>);
    });

    afterAll(()=>{
        component = null;
    });

    it('Renders properly', ()=>{
        expect(component.toJSON()).toMatchSnapshot();
    });
    
    it('Renders the home container', ()=>{
        expect(component.root.findAllByProps({id: "home"}).length).toBe(1); 
    });
    
    it('Renders the <UserList /> component', ()=>{
        expect(component.root.findAllByProps({className: "users-list"}).length).toBe(1); 
    });
    
    it('Does not render the <Loader /> component', ()=>{
        expect(component.root.findAllByProps({className: "loader"}).length).toBe(0); 
    });

});
