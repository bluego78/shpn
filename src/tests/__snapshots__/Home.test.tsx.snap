// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> when context is loaded Renders properly 1`] = `
<div
  className="content-fluid d-flex flex-column align-items-center justify-content-center"
  id="home"
>
  <div
    className="users-list"
    onScroll={[Function]}
  >
    <div
      className="badge-container"
      onClick={[Function]}
    >
      <div
        className="userbadge-container"
      >
        <div
          className="nat-flag"
        >
          <img
            alt="John Doe picture"
            src="ch.png"
          />
        </div>
        <img
          alt="John Doe picture"
          className="picture"
          src="https://randomuser.me/api/portraits/thumb/men/56.jpg"
        />
        <div
          className="fullname"
        >
          John
           
          Doe
        </div>
        <div
          className="username"
        >
          theusername
        </div>
        <div
          className="email"
        >
          test@example.com
        </div>
      </div>
    </div>
    <div
      className="badge-container"
      onClick={[Function]}
    >
      <div
        className="userbadge-container"
      >
        <div
          className="nat-flag"
        >
          <img
            alt="John Smith picture"
            src="gb.png"
          />
        </div>
        <img
          alt="John Smith picture"
          className="picture"
          src="https://randomuser.me/api/portraits/thumb/men/57.jpg"
        />
        <div
          className="fullname"
        >
          John
           
          Smith
        </div>
        <div
          className="username"
        >
          anotherusername
        </div>
        <div
          className="email"
        >
          test2@example.com
        </div>
      </div>
    </div>
  </div>
</div>
`;
